fpga_interchange/yosys_json.py

```diff
@@ -18,20 +18,44 @@
         CellInstance, Direction, Library
 from fpga_interchange.parameter_definitions import ParameterFormat
 
+BUS_INDEX_RE = re.compile(r"(?P<name>[^\s\[\]]*)(\[(?P<index>[0-9]+)\])?")
 
-def is_bus(bits, offset, upto):
+
+def is_bus(name, bits, offset, upto):
     """ Returns true if this port/net is a bus.
 
-    >>> is_bus([2], offset=0, upto=0)
+    >>> is_bus("net", [2], offset=0, upto=0)
     False
-    >>> is_bus([2, 3], offset=0, upto=0)
+    >>> is_bus("net", [2, 3], offset=0, upto=0)
     True
-    >>> is_bus([2], offset=1, upto=0)
+    >>> is_bus("net", [2], offset=1, upto=0)
     True
-    >>> is_bus([2], offset=0, upto=1)
+    >>> is_bus("net", [2], offset=0, upto=1)
     True
+    >>> is_bus("net[3]", [100], offset=3, upto=0)
+    False
 
     """
+
+    # In case of a port which is a result of the "splitnets" pass the
+    # original offset is preserved even though the port is no longer a bus.
+    #
+    # Detect it here by looking at the bit indes in brackets (which is a part
```

Comment:
typo: indes -> index
```suggestion
    # Detect it here by looking at the bit index in brackets (which is a part
```

Reply:
fixed.